web3.js/src/ed25519-program.ts

```diff
@@ -82,17 +94,18 @@ export class Ed25519Program {
 
     const instructionData = Buffer.alloc(messageDataOffset + message.length);
 
+    const index = instructionIndex != null ? instructionIndex : 0;
```

Comment:
Passing `undefined` essentially produces a zero in the buffer, so let's default to zero.

Reply:
0xffff would be better